core/src/main/scala/core/database/Query.scala

```diff
@@ -0,0 +1,28 @@
+package core.database
+
+import java.sql.Connection
+
+import core.utils.Reader
+
+case class Query[A](underlying: Reader[Connection, A]) { self =>
+  type DatabaseContext
```

Comment:
For me, it couples each query since it's created with a fixed and determined context, rather than being able to then executed a query (possible multiple time) with context(es) resolved/prepared later.

Reply:
If the user sequences ops using this monad, don't we need to enforce the use of a transaction during interpretation ? If so how can we do this without keeping the context in the query?

Reply:
Defining a `Query` and its transformation is "before" interpreting (context unknown).